lib/core/src/mqtt.rs

```diff
@@ -0,0 +1,122 @@
+use defmt::*;
+#[cfg(not(feature = "std"))]
+use heapless::String;
+use rust_mqtt::{
+    client::{client::MqttClient, client_config::ClientConfig},
+    packet::v5::reason_codes::ReasonCode,
+    utils::rng_generator::CountingRng,
+};
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "std"))]
+pub struct MqttMessage {
+    pub topic: String<48>,
+    pub payload: String<512>,
+}
+
+#[cfg(feature = "std")]
+pub struct MqttMessage {
+    pub topic: String,
+    pub payload: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ButtonMqttMessage {
+    pub task_id: u8,
+    pub status: bool,
+}
+
+pub struct HypedMqttClient<
+    'a,
+    T: embedded_io_async::Read + embedded_io_async::Write,
+    R: rand_core::RngCore,
+> {
+    pub client: MqttClient<'a, T, 5, R>,
+}
+
+pub fn initialise_mqtt_config() -> ClientConfig<'static, 5, CountingRng> {
+    let mut config = ClientConfig::new(
+        rust_mqtt::client::client_config::MqttVersion::MQTTv5,
+        CountingRng(20000),
```

Comment:
Why 20000?

Reply:
Complete honesty, idk it's the example for the library and the number is just used as the starting value for the counter to emit nonces.